src/test/java/support/fakes/FakeStorageModule.java

```diff
@@ -87,6 +91,31 @@ void register(Router router) {
 
     router.get(rootPath + "/:id").handler(this::get);
     router.delete(rootPath + "/:id").handler(this::delete);
+    router.post(rootPath + "/emulate-failure").handler(this::emulateFailure);
+  }
+
+  private void emulateFailureIfNeeded(RoutingContext routingContext) {
```

Comment:
Why is it necessary to emulate a failure with a special endpoint?

Reply:
It is used to test that when we searching for instances and an error occurred with sub-calls (e.g. to retrieve relationships or preceding-succeeding titles) then it have to be handled in some way.

Unfortunately, it can not be tested just with unit tests, because the logic is quite encapsulated, so no way to test just a piece of functionality. 

(My opinion, is that we need a refactoring for the instances endpoint).

Reply:
> It is used to test that when we searching for instances and an error occurred with sub-calls (e.g. to retrieve relationships or preceding-succeeding titles) then it have to be handled in some way.

Given that this is being tested from the outside, how does the system know to use the new endpoint rather than the relevant preceding / succeeding titles endpoint?

> Unfortunately, it can not be tested just with unit tests, because the logic is quite encapsulated, so no way to test just a piece of functionality.
> 
> (My opinion, is that we need a refactoring for the instances endpoint).

Please raise a tech debt issue for this

Reply:
> Given that this is being tested from the outside, how does the system know to use the new endpoint rather than the relevant preceding / succeeding titles endpoint?

It is not a new endpoint, it is just a sub-path to existing endpoint, so the whole path will be for example `/preceding-succeeding-titles/emulate-failure`. And steps to test a failure case might be following:
* Do some preparation (e.g. create instances/relationships);
* Enable failure emulation for the API for some time period, by sending request to the mentioned URI;
* Attempt call the API that needs to be tested;
* Verify response;
* In an `@After` method do disable of failure emulation.

Reply:
To me, that is a new endpoint.

Oh, this endpoint is used to control when other endpoints in that fake report failures. Is that a sensible understanding?

Reply:
Yes, correct. It is not ideal, I think we should consider a better approach, but it is an important case to verify it.

Reply:
> Please raise a tech debt issue for this.

Here it is https://issues.folio.org/browse/MODINV-213